src/PetriEngine/Reducer.cpp

```diff
@@ -874,40 +891,24 @@ namespace PetriEngine {
                 }
             }
 
-            if(!ok || notenabled.size() == 0) continue;
-
-            for(uint prod : place.producers)
-            {
-                if(notenabled.count(prod) == 0)
-                {
-                    ok = false;
-                    break;
-                }
-                // check that producing arcs originate from transition also
-                // consuming. If so, we know it will never fire.
-                Transition& t = getTransition(prod);
-                ArcIter it = getInArc(p, t);
-                if(it == t.pre.end())
-                {
-                    ok = false;
-                    break;
-                }
-            }
-
-            if(!ok) continue;
-
-            _ruleE++;
-            continueReductions = true;
-
-            if(placeInQuery[p] == 0)
-                parent->initialMarking[p] = 0;
+            if(!ok || notenabled.empty()) continue;
 
             bool skipplace = (notenabled.size() == place.consumers.size()) && (placeInQuery[p] == 0);
-            for(uint cons : notenabled)
+            bool E_used;
+            for(uint cons : notenabled) {
+                Transition &t = getTransition(cons);
+                auto in = getInArc(p, t);
```

Comment:
Please take a look at these warnings; the "in" variable is not used.

Reply:
That is an artifact of the inhibitor arc extension we added while fixing it, but which i wrote out of this pull request as we haven't proven it valid yet. I will do another look at it.